policyengine_uk/variables/gov/dwp/extended_childcare_entitlement/conditions/extended_childcare_entitlement_work_condition.py

```diff
@@ -0,0 +1,51 @@
+from policyengine_uk.model_api import *
+
+
+class extended_childcare_entitlement_work_eligibility(Variable):
+    value_type = bool
+    entity = Person
+    label = "Extended childcare entitlement work eligibility"
+    documentation = "Eligibility maintained for a single parent working or receiving incapacity benefit, or a couple with one or both working."
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        """
+        Calculate eligibility for free childcare based on the following conditions:
+        - Single parent: working or not working but receiving incapacity benefit
+        - Couple: both working or one working and one receiving incapacity benefit
+        """
+        # Get the benefit unit the person belongs to
+        benunit = person.benunit
+        is_adult = person("is_adult", period).astype(bool)
+
+        # Check work status (in_work)
+        in_work = person("in_work", period).astype(bool)
+
+        # Check if the person is receiving incapacity benefit (simplified condition)
+        has_incapacity_benefit = (
+            person("incapacity_benefit", period) > 0
+        ).astype(bool)
+
+        # Build conditions for Single Parent
+        is_single = (benunit.sum(is_adult) == 1).astype(bool)
+        single_working = is_single & in_work
+        single_not_working_but_eligible = (
+            is_single & ~in_work & has_incapacity_benefit
+        )
+
+        # Build conditions for Couples
+        is_couple = (benunit.sum(is_adult) == 2).astype(bool)
```

Comment:
In general should almost never need to cast
```suggestion
        is_couple = benunit.sum(is_adult) == 2
```